src/installer.ts

```diff
@@ -370,14 +382,14 @@ async function queryDistForMatch(
       throw new Error(`Unexpected OS '${osPlat}'`);
   }
 
+  if (!nodeVersions) {
+    core.debug('No dist manifest cached');
+    nodeVersions = await getVersionsFromDist();
+  }
+
```

Comment:
I believe it will be easier to invoke `getVersionsFromDist()` once without any conditions in `queryDistForMatch` function. In this case we can remove `getVersionsFromDist()` from this code:
```javascript
if (isLatestSyntax(versionSpec)) {
    nodeVersions = await getVersionsFromDist();
    versionSpec = await queryDistForMatch(versionSpec, arch, nodeVersions);
    core.info(`getting latest node version...`);
  }
```
In this case we no need to get and pass `nodeVersions` array to `queryDistForMatch` function because we use this array only for this condition.

I believe it's more clear logic.

Reply:
@vsafonkin In this case we can have a case to call dist api twice, that is why we grab all versions at the beginning and pass it to further functions.